### PR TITLE
Add button to toggle drawer

### LINKED ISSUE #389: sidebar: collapse should always be an option
Presently the sidebar can only be collapsed in small windows. We should allow users to collapse the sidebar at all times.

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by kinow**
These changes close #389

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [ ] Appropriate tests are included (unit and/or functional).
- [ ] Does not need tests (why?).
- [ ] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

File: src/styles/cylc/_drawer.scss [hunk 1]
```diff
@@ -40,3 +51,7 @@
     padding-right: 15px;
   }
 }
+
+.v-navigation-drawer--close {
+  visibility: unset !important;
```
Comment (kinow) [inline src/styles/cylc/_drawer.scss:56]: The main panel has higher z-index, so it's displayed if the drawer is still visible.

**Comment by kinow** [comment]
@hjoliver @oliver-sanders here's a possible solution; the drawer normally is toggled in two ways in Vuetify. One way is to have a toggle button outside of the drawer. Normally it's in the main content area, or in the toolbar. Cylc UI, however, does not always show a toolbar. For example, the dashboard has no toolbar visible, so it wouldn't be possible to put a toggle (or burger) button there.

The other way is to have a button in the drawer. The only issue is that when this approach is used, then the toolbar is not hidden. Instead a variant is activated when toggled, showing a "mini" toolbar (showing only icons, no text). That wouldn't work in mobile though. So I used this second option, with a variation with a floating button. It seems to work OK. If this looks like a good approach, I'll write the e2e test :+1: 

GIFrecord_2021-08-09_083415

GIFrecord_2021-08-09_083656

(looks like Dashboard is broken for mobile, will take care of that in another PR)

**Comment by hjoliver** [comment]
Nice, that looks good to me.

**Comment by wxtim** [comment]
stale

**Comment by hjoliver** [comment]
> stale

(We can now resize the sidebar away)

**Pull request closed without merging**